src/hooks.server.ts

```diff
@@ -1,6 +1,48 @@
 import type { Handle } from "@sveltejs/kit";
+import { KV_URL } from "$env/static/private";
+import { createClient as createRedisClient, type RedisClientType } from "redis";
+import { sequence } from "@sveltejs/kit/hooks";
 
-export const handle = (async ({ event, resolve }) => {
+type None = Record<never, never>;
+
+const createConnectedRedisClient = async ({
+  url,
+  useTLS = false,
+}: {
+  url: string;
+  useTLS: boolean;
+}): Promise<RedisClientType<None, None, None> | undefined> => {
+  let client: RedisClientType<None, None, None>;
+  try {
+    client = createRedisClient<None, None, None>({
+      url,
+      socket: { tls: useTLS },
+    });
+    await client.connect();
+  } catch (_) {
+    return;
+  }
+  return client;
+};
+
+const handleSetupRedisClient = (async ({ event, resolve }) => {
+  // we intentionally don't await this promise here, so that other things can happen while redis is
+  // initializing. Anything that needs redis can await the promise by calling
+  // event.locals.getConnectedRedisClient() and awaiting the result. If nothing awaits the promise
+  // then we never wait on redis connecting first.
+  const redisClientPromise = !KV_URL
+    ? Promise.reject(new Error("could not get redis client"))
+    : createConnectedRedisClient({ url: KV_URL, useTLS: !KV_URL.startsWith("redis://localhost") });
+
+  // So we don't get unhandled promise rejection warnings if nothing awaits the promise
+  redisClientPromise.catch((err) => console.error(err));
```

Comment:
If #274 is merged first, this will need to be updated to use the `logger` added in that PR.